server/container/util/src/main/java/org/apache/james/util/ReactorUtils.java

```diff
@@ -45,6 +46,10 @@
         return (element, sink) -> element.ifPresent(sink::next);
     }
 
+    public static <T, U> BiConsumer<U, SynchronousSink<T>> transformNullable(Function<U, T> mapper) {
```

Comment:
same comment as previous commit: such generic concepts are hard to name and without a good name they hurt more than they solve things

Reply:
mapNullabble?

`flux.handle(mapNullable(x::operation))` is pretty understandable to me.

In the calling code the naming is perfectly good and understandable.

Only in the utility class does it makes little sense.

I don't see how we are different from https://github.com/linagora/james-project/pull/3314#discussion_r412187266 & https://github.com/linagora/james-project/pull/3314#discussion_r412189619

Reply:
I removed `or` from the linked PR BTW.

map is a method of Monad (Functor actually) and Nullable is not a Monad ( nor a Functor).

transform was more correct, 

I still believe it's useless because you can very well hide the technical bits by extracting the handling into a method.

Reply:
I propose `transformAndPublishIfNotNull` naming.

> I still believe it's useless because you can very well hide the technical bits by extracting the handling into a method.

If I extract something, I end up with the exact ReactorUtils code snipset duplicated in every classes.